SELV3-575: Added functionality for fetching DHIS2 API datasets.

sradziszewski opened the pull request:
Added DhisDataService
Fixed MessageKeys class duplication.
Added RestTemplate configuration.
Added model classes for DHIS2 API dataset.
Added helper classes: RequestHelper and RequestParameters.
JIRA task link: [SELV3-575](https://openlmis.atlassian.net/browse/SELV3-575)

[SELV3-575]: https://openlmis.atlassian.net/browse/SELV3-575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

pwargulak commented on src/main/java/org/openlmis/integration/dhis2/dto/dhis/DataElement.java:
```diff
@@ -0,0 +1,30 @@
+@Getter
+@Setter
+@ToString
+public class DataElement {
```
I think that a knowledge about DHIS2 data model might be lost after few months. 
Please add some JavaDoc which roughly explains what the class is about - like DataElement is a description of single data value, not a calculated value itself but 'a type/kind' (?).
Think what you would like to read here after a year you code it.

pwargulak commented on src/main/java/org/openlmis/integration/dhis2/dto/dhis/DhisDataset.java:
```diff
@@ -0,0 +1,35 @@
+@Getter
+@Setter
+@ToString
+public class DhisDataset {
```
(same as DataElement)
I think that a knowledge about DHIS2 data model might be lost after few months. 
Please add some JavaDoc which roughly explains what the class is about - like DataElement is a description of single data value, not a calculated value itself but 'a type/kind' (?).
Think what you would like to read here after a year you code it.

sradziszewski replied:
I added JavaDoc.

pwargulak commented on src/main/java/org/openlmis/integration/dhis2/dto/dhis/OrganisationUnit.java:
```diff
@@ -0,0 +1,30 @@
+@Getter
+@Setter
+@ToString
+public class OrganisationUnit {
```
(same as DataElement)
I think that a knowledge about DHIS2 data model might be lost after few months. 
Please add some JavaDoc which roughly explains what the class is about - like DataElement is a description of single data value, not a calculated value itself but 'a type/kind' (?).
Think what you would like to read here after a year you code it.

sradziszewski replied:
I added JavaDoc.

pwargulak reviewed (changes requested):
Small JavaDoc and should be fine.

sradziszewski commented on src/main/java/org/openlmis/integration/dhis2/util/RequestParameters.java:
```diff
@@ -0,0 +1,47 @@
+
+public class RequestParameters {
+  private final Map<String, Object> params = Maps.newHashMap();
+
```
Empty line added.

sradziszewski commented on src/test/java/org/openlmis/integration/dhis2/util/RequestHelperTest.java:
```diff
@@ -0,0 +1,60 @@
+    assertThat(entity.getHeaders().get(HttpHeaders.AUTHORIZATION),
+            is(singletonList("Prefix r4nd0m70k3n")));
+  }
+}
```
Empty line added.

sradziszewski commented on src/test/java/org/openlmis/integration/dhis2/util/RequestParametersTest.java:
```diff
@@ -0,0 +1,47 @@
+
+    return map;
+  }
+}
```
Empty line added.

sradziszewski commented:
@pwargulak 
After consulting with @anawrotsoldevelo, I added the **code** field to the Organization Unit model.
I also changed the type of the dataSetElements field in the DhisDataSet class and added a getter to bypass the **empty wrapper** from the DHIS2 API. Example:
```
"dataSetElements": [
        {
            "dataElement": {
                "name": "Plague (Deaths < 5 yrs)",
                "id": "lXolhoWewYH"
            }
        },
        {
            "dataElement": {
                "name": "Yellow Fever (Deaths < 5 yrs)",
                "id": "USBq0VHSkZq"
            }
        }
]
```

pwargulak reviewed (approved):
Looks fine
